core/src/main/scala/com/avast/cloud/metrics/datadog/impl/GaugeImpl.scala

```diff
@@ -0,0 +1,18 @@
+package com.avast.cloud.metrics.datadog.impl
+
+import cats.effect.Sync
+import com.avast.cloud.metrics.datadog.api.{ Gauge, Tag }
+import com.timgroup.statsd.StatsDClient
+
+import scala.language.higherKinds
+
+class GaugeImpl[F[_]: Sync](statsDClient: StatsDClient, aspect: String, sampleRate: Double) extends Gauge[F] {
+  private[this] val F = Sync[F]
```

Comment:
+1 for `private[this]`

Reply:
i still don't like it but i knew you would force me to add it :D